src/tee.jl

```diff
@@ -26,14 +28,38 @@ function handle_message(demux::TeeLogger, args...; kwargs...)
     end
 end
 
-function shouldlog(demux::TeeLogger, args...)
+function shouldlog(demux::AbstractTeeLogger, args...)

```

Comment:
`shouldlog`, `min_enabled_level`, and `catch_exceptions` for `FirstMatchLogger` are all identical to `TeeLogger` so I created an abstract type.  Is it OK?